### ISSUE TITLE
app_sync mysteriously gets owned by userid 82

### ISSUE THREAD

**Issue by tomfriedhof** [EugenMayer/docker-sync#453]
I'm running into permissions issues with Unison.  The `app_sync` directory mysteriously gets owned by user `82`.  Not sure what is triggering that. 

Here is my docker-sync.yml:
```
syncs:
  docker-sync: # tip: add -sync and you keep consistent names as a convention
    src: './'
    sync_host_ip: 'auto'
    sync_userid: 33
    sync_groupid: '33'
    sync_strategy: 'native_osx'
```
I can fix the unison permissions issue by manually logging into the sync container and chown back to the `xfs` user.

**Comment by EugenMayer** [comment]
i would bet that you have a chown on your app container or similar. Please compare the actual uid of the user dockersync on the sync container - is it 33? 

I bet it is not, since the user is taken by another user already, and that it xfs ( 33 ) https://github.com/EugenMayer/docker-image-unison/blob/feature/unison-hostsync/entrypoint.sh#L27 .. so dockersync cannot use it.

Thats just a suggestion and we might need to dive deeper into this probably, e.g. just not creating dockersync at all, but rather use the existing user with the expected uid then.

Please ensure if you have a system-uid collision here, thats what the core qeustion is

**Comment by tomfriedhof** [comment]
Hi Eugene,

That seems to have fixed the issue.  I used an id that isn't on the system, and things seem to be working fine now.  Thanks.

Tom

**Comment by EugenMayer** [comment]
great!

**Comment by tomfriedhof** [comment]
I'm using the docker4drupal stack and noticed this:

https://github.com/wodby/drupal-nginx/blob/master/8/1.13/init/drupal-nginx.sh#L9

That's likely what was causing the issue.  Thanks for the help!

**Comment by rodrigoaguilera** [comment]
I experienced the same with the docker4drupal images. Tell me if you find a solution. Or maybe open an issue there to have an environment variable that can change that behavior for development.

**Comment by tomfriedhof** [comment]
Hi @rodrigoaguilera, I just needed to change my `sync_userid` to `82`, a uid that wasn't already used in the sync container.

https://github.com/wodby/docker4drupal/blob/master/docker-sync.yml#L6

**Comment by EugenMayer** [comment]
i am still not very sure what this is about. If the script does actually chown to ui 82... why did you every use 33 in you settings at all?

AFAICS that is not an issue from the docker4drupal, its a configuration issue? The uid on the target app container is 82, thats the UID you have to use in the docker-sync.yml and now other .. so either i do not get it, or simply put, you just made a mistake, probably both :) 

What did you both expect sync_users to be or to map to, the the OSX uid? if yes, you might come from the alternative soltutions like dingy or dlite .. they turn it arround and force the actual app container to adopt the OSX user uid and that is .. pretty bad since you have to adjust the image most probably to make this configurable.

With docker sync, all you need to the docker-sync is, what your app container expects, in this case, since its a FPM application and the pool is configured to use www-data, it should be www-data, so just check www-data uid on the app (fpm) container/image and use that.

Is that part just unclear in the docs?

**Comment by tomfriedhof** [comment]
The docs make sense, I realized that after I solved my issue.  I pulled my `docker-sync.yml` config from another project that I had previously setup that was using a nodejs stack, and the sync_userid in that `docker-sync.yml` was 33.  That sync strategy was using rsync, so maybe the uid doesn't make a difference with that strategy.

I found the docker4drupal `docker-sync.yml` example after you had answered my question.  Had I followed that example, I would have not run into this issue.

> With docker sync, all you need to the docker-sync is, what your app container expects, in this case, since its a FPM application and the pool is configured to use www-data, it should be www-data, so just check www-data uid on the app (fpm) container/image and use that.

The docs are clear, and your point above makes it even more clear to me now.  Thanks!  Docker-sync is awesome!
